### PR TITLE
Make sure Include is in the main configuration file when drop-in directory is used

### PR DESCRIPTION
**Pull request by Jakuje**
This implements sanity check to make sure the configuration files on RHEL9 and Fedora (and now also Ubuntu 22.04), which are placed in drop-in directory are really applied by checking the include directive is present in the main configuration file.

There are also CI fixes, refactoring and the whole new support for new Ubuntu 22.04.

### CODE REVIEW

**Comment by Jakuje** [comment]
@mattwillsher @richm can you have a look?

**Comment by mattwillsher** [comment]
It looks ok to me.

We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite

**Comment by Jakuje** [comment]
> > We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite
> 
> Yeah - probably time for a module

Thank you for the comments. You are right, the conditions are getting longer. I can think of some reshuffling specific tasks to different tasks files to make it shorter and more readable. Do you have some specific recommendation except for rewriting all of that into module?

**Comment by richm** [comment]
> > > We are getting quite a lot of logic and long conditionals now. I'm wondering how far we are off this needing a refactor or even rewrite
> > 
> > 
> > Yeah - probably time for a module
> 
> Thank you for the comments. You are right, the conditions are getting longer. I can think of some reshuffling specific tasks to different tasks files to make it shorter and more readable. Do you have some specific recommendation except for rewriting all of that into module?

Not specifically, no

**Comment by Jakuje** [comment]
> Not specifically, no

So far I only split the huge install.yml to smaller parts to get rid of some conditions. Not sure if it is much better. Any other suggestions how to make it more readable would be welcomed.

In the meantime, I added support for recent ubuntu 22.04, which also finally supports drop-in directory. I can split this work into separate PR if needed.

**Comment by Jakuje** [comment]
@richm or @mattwillsher  is there something else to address or does it look ok to you?

**Comment by mattwillsher** [comment]
Looks ok to me

File: tasks/check_fips.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Check the kernel FIPS mode
+  slurp:
```
Comment (richm) [inline tasks/check_fips.yml:3]: I've been told that there is no plan to require FQCN for built-ins in any foreseeable release of Ansible. I would suggest adding the suppression to .ansible-lint

Comment (Jakuje) [reply tasks/check_fips.yml:3]: I already added it in previous PR (hopefully right). It makes the action not fail, but still reports these issues:

https://github.com/willshersystems/ansible-sshd/blob/master/.ansible-lint#L4

Comment (richm) [reply tasks/check_fips.yml:3]: ack - lgtm

File: README.md [hunk 2]
```diff
@@ -199,10 +204,12 @@ file that this role produces.
 By default (*auto*), this list contains all the host keys that are present in
 the produced configuration file. If there are none, the OpenSSH default list
 will be used after excluding non-FIPS approved keys in FIPS mode. The paths
-are checked for presence and generated if missing. Additionally, permissions
-and file owners are set to sane defaults. This is useful if the role is used
-in deployment stage to make sure the service is able to start on the first
-attempt. To disable this check, set this to empty list.
+are checked for presence and new keys are generated if they aremissing.
```
Comment (ep69) [inline README.md:207]: Typo: `aremissing`

**Review by ep69 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Jakuje) [reply README.md]: Thanks. Fixed and rebased.

File: README.md [hunk 3]
```diff
@@ -148,24 +145,32 @@ is *true*.
...
 
 The path where the openssh configuration produced by this role should be saved.
-This is useful mostly when generating configuration snippets to Include.
+This is useful mostly when generating configuration snippets to Include from
+drop-in directory (default in Fedora and RHEL9).
+
+When this path points to system drop-in directory (defined with internal
+variable `__sshd_drop_in_dir`), the main configuration file (defined with
+internal variable `__sshd_main_config_file`) is checked to contain the
+defailt `Include` directive from `__sshd_defaults` dict.
```
[the remark this answers is not in the document]
Comment (Jakuje) [reply README.md]: Thanks. Fixed and rebased.

**Comment by Jakuje** [comment]
@mattwillsher do you want to take care of the release or should I draft something? I think it still involves the manual step of publishing the role on the galaxy.

**Comment by mattwillsher** [comment]
To summarise for the release notes:

- adds support for Ubuntu 22.04
- improved support for RHEL9 and Fedora use of drop-in config
- genera quality improvements

**Comment by Jakuje** [comment]
I would probably add points like this as there is a bit more changes since last release:

 * Support for Ubuntu 22.04 with drop-in directory (71eab11)
 * When using drop-in directory, verify the main configuration file has include directive (#178)
 * Tests improvements and general maintenance
 * Ensure the needed facts are available (e24ff31)
 * Filter out incompatible host keys on FIPS systems (#173)

https://github.com/willshersystems/ansible-sshd/compare/v0.14.1...master

**Comment by mattwillsher** [comment]
v0.15.0 released and uploaded to Galaxy. Thanks all.

**Pull request merged**